public/app/features/dashboard/components/GenAI/GenAIDashDescriptionButton.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+
+import { DashboardModel } from '../../state';
+
+import { GenAIButton } from './GenAIButton';
+import { Message, Role } from './utils';
+
+interface GenAIDashDescriptionButtonProps {
+  onGenerate: (description: string, isDone: boolean) => void;
+  dashboard: DashboardModel;
+}
+
+const DESCRIPTION_GENERATION_STANDARD_PROMPT =
+  'You are an expert in Grafana dashboards.' +
+  'Your goal is to write short, descriptive, and concise dashboards description using the dashboard panels title and descriptions. ' +
+  'The description should be shorter than 140 characters.';
+
+export const GenAIDashDescriptionButton = ({ onGenerate, dashboard }: GenAIDashDescriptionButtonProps) => {
```

Comment:
Not sure how scalable this current approach is (creating a dedicated component for every use case as a place to store the prompt / getMessages function - it is probably fine for now but we'll want to think about modifying this structure as we scale these features